### PR TITLE
repack step 1

### PR DESCRIPTION
**Pull request by tduretz**
* allow for tuples as input and name tuples as output
* create specific environment for example
* remove Plots, SparseDiffTools and Symbolics from deps
* update function docs: default parameters are stated + tuple call examples
* update README.md

### CODE REVIEW

File: src/Diffusion1D/Diffusion1D_Gaussian.jl [hunk 1]
```diff
@@ -37,4 +37,11 @@ function Diffusion1D_Gaussian(x;
     hessu  = ForwardDiff.hessian(f_cl, x)
     s      = gradu[2] - params.K*(hessu[1,1])
     return (u=u, ∇u=gradu, s=s)
+end
+
+function Diffusion1D_Gaussian(coords::Tuple;
```
Comment (albert-de-montserrat) [inline src/Diffusion1D/Diffusion1D_Gaussian.jl]: Does this work for `NamedTuple`s? 

```julia-repl
julia> a=(;x=1, y=2)
(x = 1, y = 2)

julia> typeof(a) isa Tuple
false

julia> typeof(a) <: Tuple
false
```

Comment (tduretz) [reply src/Diffusion1D/Diffusion1D_Gaussian.jl]: thanks a lot for suggesting the change to `::Union{Tuple, NamedTuple}` !

File: examples/benchmarks/ExamplePoisson1D.jl [hunk 2]
```diff
@@ -0,0 +1,123 @@
+using Plots # need to install Plots package 
+import LinearAlgebra: norm
+
+𝑢(𝑥)    =  cos(𝑥) # manufactured solution
+∇𝑢(𝑥)   = -sin(𝑥) 
+𝑞(𝑥,𝛽)  = -𝛽*sin(𝑥) 
+∇𝑞(𝑥,𝛽) = -𝛽*cos(𝑥)
+
+function main(ncx, method)
```
Comment (albert-de-montserrat) [inline examples/benchmarks/ExamplePoisson1D.jl]:
```suggestion
@views function main(ncx, method) 
```

File: examples/benchmarks/ExamplePoisson1D.jl [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        dudx[i] = ∇𝑢(xve[i])
+        q_ex[i] =  𝑞(xve[i], β)
+        q[i]    =  𝑞(xve[i], β)
+    end
+
+    # Pseudo-transient integration
+    Δτ = Δx^2/β/2.1 
+    θ  = 0.038/Δx/1.6 
+
+    if method==:Spotz         # Discretisation of Spotz et al., 1995
```
Comment (albert-de-montserrat) [inline examples/benchmarks/ExamplePoisson1D.jl]:
```suggestion
    if method===:Spotz         # Discretisation of Spotz et al., 1995
```

File: examples/benchmarks/ExamplePoisson1D.jl [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            nF          = norm(F)/sqrt(length(F))
+            nF < 1e-12 ? break : nothing
+            # mod(iter, 100)==0 ? @printf("Iter %05d --- abs. res. = %1.4e\n", iter, nF) : nothing 
+            # Rate update
+            ∂u∂τ       .= (1-θ).*∂u∂τ .+ F 
+            # Variable update
+            u[2:end-1] .+= Δτ*∂u∂τ
+        end
+
+    elseif method==:Abide # Discretisation of Abide et al., 2020
```
Comment (albert-de-montserrat) [inline examples/benchmarks/ExamplePoisson1D.jl:77]:
```suggestion
    elseif method===:Abide # Discretisation of Abide et al., 2020
```

**Review by albert-de-montserrat — CHANGES_REQUESTED** [review]

**Comment by luraess** [comment]
LGTM - do not have time to test it, but thanks for incorporating these new changes!

**Review by luraess — APPROVED** [review]
LGTM - I do not have time to test it right now, but thanks for incorporating these new changes!

**Comment by tduretz** [comment]
@JuliaRegistrator register

**Pull request merged**